pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -350,7 +352,12 @@ public <T> CompletableFuture<Producer<T>> createProducerAsync(ProducerConfigurat
             return lookup.getSchema(TopicName.get(conf.getTopicName()))
                     .thenCompose(schemaInfoOptional -> {
                         if (schemaInfoOptional.isPresent()) {
-                            autoProduceBytesSchema.setSchema(Schema.getSchema(schemaInfoOptional.get()));
+                            SchemaInfo schemaInfo = schemaInfoOptional.get();
+                            if (schemaInfo.getType() == SchemaType.PROTOBUF) {
+                                autoProduceBytesSchema.setSchema(new GenericAvroSchema(schemaInfo));
```

Comment:
Will it be better if we  make the `AutoConsumeSchema` support `PROTOBUF`? 

https://github.com/apache/pulsar/blob/9a85dea53a6d0896cc4ba1ceb1ab11a47d5d65da/pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AutoConsumeSchema.java#L223-L226

Reply:
The Auto_ConsumerSchema can not support `PROTOBUF`.